### ISSUE TITLE
Renaming a tag to an existing tag creates tag duplicates on entries

### ISSUE THREAD

**Issue by dsalo** [shaarli/Shaarli#2084]
I had two tags "harassment" and "abuse" that I wanted to consolidate. (I apologize for this example, I sometimes teach very depressing stuff.) I renamed "harassment" to "abuse" via the tag-rename tool (which I love, thank you for that). Any link that was tagged with both terms now has two "abuse" tags, as seen in the screenshot below.

![Screenshot 2024-05-10 at 14-08-09 Search abuse - Salo linkspam](https://github.com/shaarli/Shaarli/assets/272370/1703ce6f-0f02-4888-927e-7b753a580f78)

This doesn't seem to break anything -- tag-searching "abuse" still works as expected, I checked -- but it clutters the display a bit. Possible to have the tag-replace process check each link to see if the new tag is already there?

**Comment by kcaran** [comment]
I was surprised that this was occurring, because I had submitted a pull request to check for duplicates before adding tags awhile ago:

https://github.com/shaarli/Shaarli/pull/1918

But I only updated the addTag() function, not renameTag(). I've submitted a pull request to make renameTag() effectively merge the old and new tags if both exist on a bookmark.

I had to rewrite most of ManageTagControllerTest. It heavily used stubs and I don't think it was actually testing the renaming of the tags. The new version fails (as expected) if you run it against the current bookmark code.
